composer/conftest.py

```diff
@@ -13,10 +13,13 @@
 # limitations under the License.
 
 
-import airflow.utils.db
 import pytest
 
 
-@pytest.fixture(autouse=True, scope="session")
-def initalize_airflow_database():
-    airflow.utils.db.initdb()
+# this fixture initializes the Airflow DB once per session
+# it is used by DAGs in both the blogs and workflows directories
+@pytest.fixture(scope="session")
```

Comment:
It would increase the time, but I wonder if it would make sense to make all of these fixtures per-function scope? That way each test always starts with a fresh db?

Reply:
I experimented on and off with having this be at a per function basis, but was also experimenting with a lot of other things at the same time. Now that I've worked out the rest of the structure, I'll try a per-function basis again and see if

- it works as it should and there's no weird collisions (I think I may have been running into that before)
- it doesn't extend the time the tests take to a bazillion hours long

Reply:
Okay - I tried it, the good news is that it doesn't increase the amount of time the tests take by more than a few seconds. The bad news is that with the way the tests are running in kokoro, resetting the Airflow DB every test seems to cause flaky errors. I just did a few re-runs now to verify that the failures are flaky. I can't replicate them locally, which also makes me think that it could have to do with running multiple Python versions in parallel.

First try errors

- [3.6](https://source.cloud.google.com/results/invocations/97b6af32-8110-49c4-a2b2-419f21cb1ad9/targets)
- [3.7](https://source.cloud.google.com/results/invocations/96aeacf8-0b9e-4ce8-be39-35b5c2fb8b8b/targets)

Second try errors
- [3.6](https://source.cloud.google.com/results/invocations/1834c473-2640-4ef2-9aaf-16dca04168d6/targets)

Third try errors
- [3.6](https://source.cloud.google.com/results/invocations/8801d597-a6d9-44f1-a1e1-4c11c53c0556)
- [3.7](https://source.cloud.google.com/results/invocations/d2fbbb6d-4894-4791-9eed-23f1d8048094)
- [3.8](https://source.cloud.google.com/results/invocations/deac8cc3-ac9c-46a7-995b-2f98fc424ca4)

What I'm now feeling a little uncertain about is if the DBs across sessions are truly unique or if we are potentially running the risk of collisions there as well. I'm going to switch it back to the session scope and do a few re-runs to see if deletions in one fixture potentially cause problems in a different version's test run

Reply:
Alright - I just ran it three times in a row at the session scope and it passed. I just pushed a test that should fail and want to also make sure that it fails consistently.

Reply:
First try: expected failures 
Second try: technically this was when I added the noxfile config, but still got the expected failures
Third try: expected failures